File: knowledge/scaffolds/cli/python-click/src/mycli/commands.py
"""Subcommand implementations."""

import click


@click.command()
@click.argument("name")
@click.option("--greeting", "-g", default="Hello", help="Greeting to use.")
@click.pass_context
def greet(ctx, name, greeting):
    """Greet someone by name."""
    msg = f"{greeting}, {name}!"
    if ctx.obj["verbose"]:
        click.echo(f"[verbose] greeting={greeting!r}, name={name!r}")
    click.echo(msg)


@click.command()
@click.pass_context
def info(ctx):
    """Show application information."""
    from mycli import __version__

    click.echo(f"mycli v{__version__}")
    click.echo(f"Verbose: {ctx.obj['verbose']}")

File: knowledge/scaffolds/cli/rust-clap/src/commands.rs
use anyhow::Result;

pub fn greet(name: &str, greeting: &str, verbose: bool) -> Result<()> {
    if verbose {
        eprintln!("[verbose] greeting={greeting:?}, name={name:?}");
    }
    println!("{greeting}, {name}!");
    Ok(())
}

pub fn info(verbose: bool) -> Result<()> {
    let version = env!("CARGO_PKG_VERSION");
    println!("mycli v{version}");
    println!("Verbose: {verbose}");
    Ok(())
}
